src/firebolt/service/binding.py

```diff
@@ -66,6 +69,9 @@ def get_database_bound_to_engine(self, engine: Engine) -> Optional[Database]:
         """Get the Database to which an engine is bound, if any."""
         try:
             binding = self.get_many(engine_id=engine.engine_id)[0]
+        except IndexError:
+            return None
+        try:
```

Comment:
Is this intended to be a part of this change? If so, can we add tests for it?

Reply:
Sure I added a couple tests to cover